ecosystem-ci.ts

```diff
@@ -23,9 +24,18 @@ cli
 	.option('--commit <commit>', 'vite commit sha to use')
 	.option('--release <version>', 'vite release to use from npm registry')
 	.action(async (suites, options: CommandOptions) => {
+		if (
+			options.branch === 'main' &&
+			options.repo === 'vitejs/vite' &&
+			!options.commit
+		) {
+			const sha = await getPermanentRef(options.repo, options.branch)
+			if (sha) {
+				options.commit = sha
+			}
+		}
 		if (options.commit) {
 			const url = `https://pkg.pr.new/vite@${options.commit}`
-			//eslint-disable-next-line n/no-unsupported-features/node-builtins
 			const { status } = await fetch(url)
 			if (status === 200) {
 				options.release = url
```

Comment:
By this code, we will sometimes get a prebuilt package and sometimes don't, depending on when the last commit was pushed. I think we should avoid that if possible so that there aren't differences among the runs.

Reply:
i see, can you show me an instance where we give the prebuilt package? i'd like to see how can i manipulate the code to perform well on those edge-cases.

i wonder if putting `if (options.commit && !options.release)` would resolve that edge case or no? would love to hear your thoughts.

Reply:
The example case of using a prebuilt package is:

1. I push a commit in the main branch at 10:30.
2. The CI in Vite repo build the package and uploads it at 10:32.
1. The ecosystem-ci triggers at 10:35.
1. The ecosystem-ci uses the prebuilt package.

The example case of not using a prebuilt package is:

1. I push a commit in the main branch at 10:30.
1. The ecosystem-ci triggers at 10:31.
1. The ecosystem-ci does not use a prebuilt package.
2. The CI in Vite repo build the package and uploads it at 10:32.

> i wonder if putting `if (options.commit && !options.release)` would resolve that edge case or no?

I think that doesn't resolve the case.